src/pseudopeople/configuration/validator.py

```diff
@@ -0,0 +1,133 @@
+from typing import Dict, List, Union
+
+import numpy as np
+from vivarium.config_tree import ConfigTree, ConfigurationKeyError
+
+from pseudopeople.configuration import Keys
+
+
+def validate_user_configuration(user_config: Dict, default_config: ConfigTree) -> None:
+    """
+    Validates the user-provided configuration. Confirms that all user-provided
+    keys exist in the default configuration. Confirms that all user-provided
+    values are valid for their respective noise functions.
+    """
+    for form, form_config in user_config.items():
+        default_form_config = _get_default_config_node(default_config, form, "form")
+        for key in form_config:
+            _get_default_config_node(default_form_config, key, "configuration key", form)
+
+        default_row_noise_config = default_form_config[Keys.ROW_NOISE]
+        default_column_noise_config = default_form_config[Keys.COLUMN_NOISE]
+
+        for noise_type, noise_type_config in form_config.get(Keys.ROW_NOISE, {}).items():
+            default_noise_type_config = _get_default_config_node(
+                default_row_noise_config, noise_type, "noise type", form
+            )
+            _validate_noise_type_config(
+                noise_type_config, default_noise_type_config, form, noise_type
+            )
+            # TODO: validate omissions = [0, 0.5]
```

Comment:
What's this TODO waiting on?

Reply:
Oh yeah, this is an actual validation we know from the docs (where the omissio probability must be between 0 and 0.5.

Reply:
I just moved an existing todo to this location. It still remains to validate config values, but that was out of scope for this ticket